src/sinks/statsd.rs

```diff
@@ -91,28 +74,44 @@ impl StatsdSvc {
             .bytes(1300)
             .events(1000)
             .timeout(1)
-            .parse_config(config.batch)?;
-        let namespace = config.namespace.clone();
+            .parse_config(self.batch)?;
+        let namespace = self.namespace.clone();
 
-        let client = Client::new(config.address)?;
+        let (client, healthcheck) = match &self.mode {
+            Mode::Tcp(ref config) => {
```

Comment:
`match &self.mod` already take reference, is `ref` still required?

Reply:
No, thanks for spotting it!